process.go

```diff
@@ -69,12 +69,14 @@ func (p *process) Kill(ctx context.Context, s syscall.Signal) error {
 }
 
 func (p *process) Wait(ctx context.Context) (uint32, error) {
-	eventstream, err := p.task.client.EventService().Subscribe(ctx, &eventsapi.SubscribeRequest{
+	cancellable, cancel := context.WithCancel(ctx)
+	eventstream, err := p.task.client.EventService().Subscribe(cancellable, &eventsapi.SubscribeRequest{
 		Filters: []string{"topic==" + runtime.TaskExitEventTopic},
 	})
 	if err != nil {
 		return UnknownExitStatus, err
 	}
+	defer cancel()
```

Comment:
i think this defer can be moved to right after its creation so we cancel even when UnknownExitStatus is returned.